hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeySetTimesRequestWithFSO.java

```diff
@@ -96,6 +97,8 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
         throw new OMException("Key not found. Key:" + key, KEY_NOT_FOUND);
       }
       omKeyInfo = keyStatus.getKeyInfo();
+      // HDDS-setting Key name back to Ozone Key before updating cache value.
+      omKeyInfo.setKeyName(OzoneFSUtils.getFileName(key));
```

Comment:
Good find.

1. Should we use helper function defined in `OMFileRequest` and used at other places. Similar to https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileRequest.java#L508

2. Can you please add unit test for it?

Reply:
I added a unit test that I had written while trying. I hope its fine.

Reply:
sure thanks a lot @sadanand48

Reply:
> 1. Should we use helper function defined in `OMFileRequest` and used at other places. Similar to https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileRequest.java#L508
@hemantk-12 I will work on another PR containing unit test to validate if the fileTable & DirTable cache has the correct value being set for all OMRequest

Reply:
Good catch! I wonder if we should fix OMFileRequest.getOMKeyInfoIfExists() instead because it's so confusing when does the keyName has the full prefix and when does not. I don't think we handle it consistently across the board.

I am fine with the stopgap fix now but i think we should revisit the handling of keyName and fileName.

Reply:
I guess that has been done to keep it same as OBS & legacy buckets which has the full key name

Reply:
> Adding a Helper function might not be helpful for other devs, having a unit test will have a failed CI.

If `OMFileRequest` is supposed to be use to access tables, then I think it is better to do that way. Otherwise as @jojochuang  said, we should remove `keyName` update from `OMFileRequest.getOMKeyInfoIfExists()`.

In the end, I think we need to revisit the access of `keyTable` (may be `fileTable` and `dirTable` as well). It is supposed to through `OMFileRequest`, it should be used for both read and write.

Reply:
We can do this refactoring as part of another PR.

Reply:
Agree. It doesn't have to part of this PR. Please create a jira to revisit and may be refactor `OMFileRequest`.